queue_downloads.py

```diff
@@ -0,0 +1,301 @@
+#!/usr/bin/env python3
+ 
+import argparse
+from datetime import datetime
+from getpass import getpass
+import json
+from time import sleep
+import requests
+
+
+def login(base_url: str, authenticator: str, username: str, password: str) -> str:
+    """
+    Args:
+        base_url (str): URL for DataGateway without path.
+        authenticator (str): Authentication mechanism to use.
+        username (str): Username to use.
+        password (str): Password to use.
+
+    Raises:
+        RuntimeError: If a status code other than 200 is returned.
+
+    Returns:
+        str: ICAT session id.
+    """
+    url = f"{base_url}/topcat/user/session"
+    data = {"plugin": authenticator, "username": username, "password": password}
+    response = requests.post(url=url, data=data)
+    if response.status_code != 200:
+        raise RuntimeError(response.text)
+
+    return json.loads(response.content)["sessionId"]
+
+
+def queue_all_files(
+    base_url: str,
+    session_id: str,
+    input_file: str,
+    transport: str,
+    file_name: str,
+    email: str,
+) -> "list[int]":
+    """Reads from `file_name` and submits Downloads for every 10,000 listed files.
+
+    Args:
+        base_url (str): URL for DataGateway without path.
+        session_id (str): ICAT session id.
+        input_file (str):
+            File containing newline delimited filepaths for the requested data.
+        transport (str): Transport mechanism/destination to use.
```

Comment:
This should probably be transport mechanism/access method to match the two terminologies used.